benchmark/distributed/master.js

```diff
@@ -0,0 +1,193 @@
+      arg => (arg ? fs.readFileSync(path.resolve(__dirname, arg)) : undefined),
+  })
+  .options('cert', {
+    alias: 'c',
+    describe: 'Path to file that contains SSL certificate',
+    type: 'string',
+    coerce:
+      arg => (arg ? fs.readFileSync(path.resolve(__dirname, arg)) : undefined),
+  })
+  .help()
+  .alias('help', 'h')
+  .argv;
+
+const {
+  address,
+  report: reportInterval,
+  key,
+  cert,
+} = args;
+
+const [error, parsed] = parseAddress(address);
+if (error) {
+  console.error(`Invalid master address: ${error.message}`);
+  process.exit(1);
+}
+
+const { transport, listen: listenAddress } = parsed;
+
+let startTimeHR = null;
+let workersReportBuffer = [];
+const workersReports = [];
+
+const workers = [];
+let server = null;
+let master = null;
+
+let serverAddress = null;
+
+const application = {
+  master: {
+    registerServer: (connection, address, callback) => {
+      if (server) {
+        callback('Server is already registered');
+      }
+      server = connection;
+      serverAddress = address;
+      workers.forEach((worker) => {
+        worker.emitRemoteEvent('worker', 'connect', [serverAddress]);
+      });
+      connection.on('event', (iface, event, args) => {
+        if (iface !== 'master') {
+          return;
+        }
+        if (event === 'report') {
+          server.connections = args[0];
+        }
+      });
+      callback(null);
+    },
+
+    registerWorker: (connection, callback) => {
+      workers.push(connection);
+      if (serverAddress) {
+        connection.emitRemoteEvent('master', 'connect', [serverAddress]);
+      }
+
+      connection.on('event', (iface, event, args) => {
+        if (iface !== 'master') {
+          return;
+        }
+        if (event === 'report') {
+          workersReportBuffer.push(args);
+        }
+      });
+      callback(null);
+    },
+
+    deregisterWorker: (connection, callback) => {
+      const index = workers.findIndex(conn => conn === connection);
+      if (index === -1) {
+        callback('Not registered');
+        return;
+      }
+      workers.splice(index, 1);
+      callback(null);
+    },
+  },
+};
+
+const config = {
+  applications: [new jstp.Application('master', application)],
+  key,
+  cert,
+};
+
+// Prevent Node.js from throwing DEPTH_ZERO_SELF_SIGNED_CERT.
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
+
+master = jstp[transport].createServer(config);
+
+master.listen(listenAddress, () => {
+  console.log(`Master server listening on ${address}\n`);
+  startTimeHR = process.hrtime();
+
+  let intervalTimerHR = process.hrtime();
+  const report = () => {
+    const timeFromStartHR = process.hrtime(startTimeHR);
+    const timePassedHR = process.hrtime(intervalTimerHR);
+
+    if (!workersReportBuffer.length) {
+      return;
+    }
+
+    const timeFromStart = timeFromStartHR[0] * 1e9 + timeFromStartHR[1];
+    const timePassed = timePassedHR[0] * 1e9 + timePassedHR[1];
+
+    const samples = workersReportBuffer.map(
```

Comment:
Why not just split them on event handling? (As I see it you wanted to localize argument handling hence placed it here but I think it's better to handle them right away as this function has nothing to do with the format workers present their data with, also with new event handling api you will have all arguments already separated upon handling).